### PR TITLE
Add documentation for installation of robotology-superbuild using conda-forge provided dependencies

### PR DESCRIPTION
**Pull request by traversaro**
The documentation mirrors the steps used in the CI. At the moment the documentation is not linked in the README to avoid confusing new users, but I preferred to add it so I can point interested user that want to experiment with conda-forge to some exhaustive documentation, as now it is possible to install everything with no workarounds after https://github.com/robotology/robotology-superbuild/pull/603 .

Fix https://github.com/robotology/robotology-superbuild/issues/477 .

### CODE REVIEW

**Review by Nicogene — APPROVED** [review]

File: doc/conda-forge.md [hunk 1]
```diff
@@ -0,0 +1,113 @@
+# conda-forge based robotology-superbuild Installation
+
+[`conda`](https://docs.conda.io/en/latest/) is a package manager (originally for Python, but effectively language agnostic) that works on Linux, macOS and Windows. 
+
+[`conda-forge`](https://conda-forge.org/)  is a community-mantained channel for the `conda` package manager that provides many dependencies useful for scientific software, in particular all the one that are required by the robotology-superbuild .
+
+## conda-forge overview
+
+The **advantages** of the use of conda and conda-forge are: 
+* **Updated dependencies:** conda-forge tipically has relatively recent version of the dependencies, so if you want to use a recent vtk or pcl (or event a recent compiler) on an old Linux distro such as Ubuntu 16.04, you can. This permits us to support older distributions even when the packages that install via apt are relatively old.
```
Comment (S-Dafarra) [inline doc/conda-forge.md:10]: Maybe it is worth adding a link to ``vtk`` or ``pcl``, also for the newcomers who may get lost on this point

Comment (S-Dafarra) [reply doc/conda-forge.md:10]: Or maybe also just a word on what they are.

File: doc/conda-forge.md [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+
+On **Windows**, run:
+~~~
+cd robotology-superbuild
+mkdir build
+cmake -G"Visual Studio 16 2019" ..
+cmake --build . --config Release 
+~~~
+
+**IMPORTANT: conda-forge does not provide Debug version of its libraries, so in Windows you can't compile in Debug mode if you are using conda-forge.**
```
Comment (S-Dafarra) [inline doc/conda-forge.md:99]: Maybe it is worth adding it to the disadvantages

**Review by S-Dafarra — APPROVED** [review]
It looks pretty nice. I also learned more from reading it, so I guess it serves the purpose.

File: doc/conda-forge.md (hunk 1 above)
Comment (S-Dafarra) [reply doc/conda-forge.md:10]: Actually, I did not notice this was in a file specific to ``conda``, so I guess this is not really necessary. Feel free to ignore this comment.

File: doc/conda-forge.md (hunk 2 above)
Comment (traversaro) [reply doc/conda-forge.md:99]: Good point.

File: doc/conda-forge.md (hunk 1 above)
Comment (traversaro) [reply doc/conda-forge.md:10]: No, you are right, expanding a bit the sentence and add a link to it is not much effort,  and definitely improves the text.

**Comment by traversaro** [comment]
I addressed your comments @S-Dafarra . Let's merge it, if necessary we can improve it in the future.

**Comment by diegoferigo** [comment]
I saw this issue right after it was merged. An alternative (and yet experimental) way to create a conda environment is through `micromamba`. As a refernce, here below some steps I recently used in a Dockerfile:

```bash
export CONDA_PREFIX=/conda
export CONDA_PYTHON_VERSION=3.8
export MAMBA_ROOT_PREFIX=$CONDA_PREFIX/.mamba

cd /usr/local
wget -qO- https://micromamba.snakepit.net/api/micromamba/linux-64/latest | tar -xvj bin/micromamba
eval "$(micromamba shell hook -s bash)"
micromamba create -y -p $CONDA_PREFIX python==$CONDA_PYTHON_VERSION mamba -c conda-forge
micromamba activate $CONDA_PREFIX
conda config --system --add channels conda-forge
conda config --system --set channel_priority strict
```

On a new shell, you can use the following function (and store it where you prefer) to enable the environment:

```bash
function activate_conda() {
    export CONDA_PREFIX=/conda
    export MAMBA_ROOT_PREFIX=$CONDA_PREFIX/.mamba

    eval "$(micromamba shell hook -s bash)"
    micromamba activate $CONDA_PREFIX
}
```

**Pull request merged**